geocomp/point_visibility/point_visibility.py

```diff
@@ -92,11 +115,22 @@ def point_visibility_with_points(point_list: list) -> list:
         point_list.append(point_list[-1])
 
     origin_point = point_list[0]
-    segment_list = [Segment(x1, x2) for x1, x2
-                    in zip(islice(point_list, 1, None, 2), islice(point_list, 2, None, 2))]
-    
+    segment_list = []
+    for x1, x2 in zip(islice(point_list, 1, None, 2), islice(point_list, 2, None, 2)):
```

Comment:
make this `for` inside point_visibility?

```suggestion
    for idx, segment in enumerate(segment_list):
        if comparison_function(segment.init, segment.to) > 0:
            segment.init, segment.to = segment.to, segment.init

```